### PR TITLE
W-11210718: Fixed Union of Arrays validation.

### PR DESCRIPTION
**Pull request by camilalrocca**

### CODE REVIEW

File: src/test/munit/parameters/query-consistency/query-consistency-validation-test-suite.xml [hunk 1]
```diff
@@ -130,5 +130,157 @@
...
+	</munit:test>
+
+	<munit:test name="validateNonNullableUnionNullableArraysParams-QueryParams">
+		<munit:enable-flow-sources>
+			<munit:enable-flow-source value="query-consistent-validation-main"/>
+			<munit:enable-flow-source value="get:\testQueryParams:query-consistent-validation-config"/>
+		</munit:enable-flow-sources>
+		<munit:execution>
+			<http:request method="GET" config-ref="http-requester-simple"
+						  path="/api/testQueryParams?nonNullableUnionOfNullableArraysParams=null">
```
Comment (dcourtil) [inline src/test/munit/parameters/query-consistency/query-consistency-validation-test-suite.xml:274]: Although `=null` is the same as nothing, I would leave this one as the rest (without it) for consistency between test cases

**Review by dcourtil — APPROVED** [review]

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java [hunk 2]
```diff
@@ -121,7 +123,8 @@ private static void validate(String paramKey, Parameter expected, String paramVa
     }
   }
 
-  private static boolean isNullable(Parameter parameter) {
-    return parameter.validate(null);
+  private static boolean isNull(Collection<?> paramValues) {
+    return paramValues == null ||
+        paramValues.size() == 1 && paramValues.stream().allMatch(value -> value == null || "null".equals(value));
```
Comment (iglosiggio) [inline src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java:128]: This is my interpretation of the code:
> It `isNull` either if the value for the collection is `null` or if the collection has exactly one value whose value is `null` or "null"

Is this correct? (i'm probably missing some context, sorry for that)
The usage of `.stream()` makes me think that we were expecting more than one value to be tested here, but the restriction `paramValues.size()` explicitly states that we want exactly one parameter to be tested.

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java [hunk 3]
```diff
@@ -52,7 +52,9 @@ private static ValidatedQueryParams validateQueryParams(Map<String, Parameter> q
       List<String> values = incomingQueryParams.getAll(paramKey);
 
       if (!values.isEmpty()) {
-        if (parameterDefinition.isRepeat() || parameterDefinition.isArray()) {
+        if (isNull(values) && parameterDefinition.isNullable()) {
+          validateQueryParam(queryParamsCopy, paramKey, parameterDefinition, values.get(0));
```
Comment (iglosiggio) [inline src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java:56]: `isNull(null)` currently evaluates to `true`. That will cause a `NullPointerException` if we try to run `values.get(0)`

**Review by iglosiggio — COMMENTED** [review]
The changes look pretty good. I've yet to read the tests (will do after lunch).

I've left a comment about a possible NPE (dimiss it if you think it's impossible) and a function definition that looks a bit unclear to me.

Anyways, nice job :)

**Review by iglosiggio — APPROVED** [review]

**Review by dcourtil — APPROVED** [review]

**Pull request merged**